### ISSUE TITLE
calls to .Select, .Filter, .OrderBy to every IGraphService*CollectionRequest should not add the parameter if the argument is null or empty

### ISSUE THREAD

**Issue by Licantrop0** [microsoftgraph/msgraph-sdk-dotnet-core#196]
Consider this code:

```
public Task<IGraphServiceGroupsCollectionPage> GetGroupsAsync(string select, string filter, string orderBy, string search)
{
   var request = _graphClient.Users
      .Request()
      .Select(select)
      .Filter(filter)
      .OrderBy(orderBy);

   var url = request.GetHttpRequestMessage().RequestUri;
   return request.GetAsync();
}
```

in case any of the query parameter is null or empty, the Request Uri still contains the queryString key, e.g.:
`https://graph.microsoft.com/v1.0/users?$select=id, displayName&$filter=&$orderby=`

Empty string cases should be handled and the generated URL should not contain the parameter, otherwise the required code to generate a clean RequestUri is very verbose:

```
var request = _graphClient.Users
    .Request();
if (!select.IsNullOrEmpty())
    request.Select(select);
if (!filter.IsNullOrEmpty())
    request.Filter(filter);
if (!orderBy.IsNullOrEmpty())
    request.OrderBy(orderBy);
``` 
[AB#7608](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7608)

**Comment by andrueastman** [comment]
@Licantrop0 Thanks for raising this issue. 
Is this issue arising from a change that happened due to upgrading the SDK from an older version?

**Comment by Licantrop0** [comment]
@andrueastman I'm not sure, it's a behavior that I'm observing on a newly developed app because of a specific use case to display the RequestUri to the end user.

The graph call goes through well even if the querystring param values are empty.

**Comment by andrueastman** [comment]
This is now closed via #255 and available through https://www.nuget.org/packages/Microsoft.Graph.Core/2.0.0-preview.12
